wen-restart/src/wen_restart.rs

```diff
@@ -963,16 +996,40 @@ pub fn wait_for_wen_restart(config: WenRestartConfig) -> Result<()> {
                     my_heaviest_fork: Some(heaviest_fork),
                 }
             }
-            WenRestartProgressInternalState::HeaviestFork { new_root_slot } => {
-                aggregate_restart_heaviest_fork(
-                    &config.wen_restart_path,
-                    config.wait_for_supermajority_threshold_percent,
-                    config.cluster_info.clone(),
-                    config.bank_forks.clone(),
-                    config.exit.clone(),
-                    &mut progress,
-                )?;
-                WenRestartProgressInternalState::HeaviestFork { new_root_slot }
+            WenRestartProgressInternalState::HeaviestFork {
+                new_root_slot,
+                new_root_hash,
+            } => {
+                if config.cluster_info.id() == config.wen_restart_coordinator {
+                    config
+                        .cluster_info
+                        .push_restart_heaviest_fork(new_root_slot, new_root_hash, 0);
+                    // TODO(wen): remove this aggregate.
```

Comment:
Why can't this aggregate be simply removed?

Reply:
It's mostly because I don't want the HeaviestForkAggregate be dead code, we need to aggregate at the coordinator later.